mobs/nakki_boss/BossLilypad.cs

```diff
@@ -0,0 +1,194 @@
+using Godot;
+using Godot.Collections;
+using System.Linq;
+
+using Jakojaannos.WisperingWoods.Characters.Player;
+using Jakojaannos.WisperingWoods.Util.Editor;
+
+namespace Jakojaannos.WisperingWoods;
+
+[Tool]
+public partial class BossLilypad : Node2D {
+	[Export] public float UnderwaterTime { get; set; } = 1.5f;
+	[Export] public float SinkAnimationSpeed { get; set; } = 1.0f;
+	[Export] public float ShakeAnimationSpeed { get; set; } = 1.0f;
+	[Export] public Array<string> Tags { get; set; } = [];
+
+	[Export]
+	public bool SunkenByDefault {
+		get => _sunkenByDefault;
+		set {
+			_sunkenByDefault = value;
+			Visible = !SunkenByDefault;
+		}
+	}
+	private bool _sunkenByDefault = false;
+
+
+	[Export]
+	[ExportGroup("Prewire")]
+	[MustSetInEditor]
+	public AnimationPlayer AnimationPlayer {
+		get => this.GetNotNullExportPropertyWithNullableBackingField(_animationPlayer);
+		set => this.SetExportProperty(ref _animationPlayer, value);
+	}
+	private AnimationPlayer? _animationPlayer;
+
+	[Export]
+	[MustSetInEditor]
+	public CollisionShape2D Blocker {
+		get => this.GetNotNullExportPropertyWithNullableBackingField(_blocker);
+		set => this.SetExportProperty(ref _blocker, value);
+	}
+	private CollisionShape2D? _blocker;
+
+	private Timer UnderwaterTimer {
+		get => this.GetNotNullExportPropertyWithNullableBackingField(_underwaterTimer);
+		set => this.SetExportProperty(ref _underwaterTimer, value);
+	}
+	private Timer? _underwaterTimer;
+
+	private Timer ShakeTimer {
+		get => this.GetNotNullExportPropertyWithNullableBackingField(_shakeTimer);
+		set => this.SetExportProperty(ref _shakeTimer, value);
+	}
+	private Timer? _shakeTimer;
+
+
+	private Player? _player;
+	private bool _isDoneShaking;
```

Comment:
I think this doesn't need to be an explicit bool:
```suggestion
	private bool IsDoneShaking => ShakeTimer.IsStopped();
```